src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -178,19 +186,24 @@ public JObject Header
         /// Gets the 'value' of the 'JWT ID' claim { jti, ''value' }.
         /// </summary>
         /// <remarks>If the 'JWT ID' claim is not found, an empty string is returned.</remarks>
-        public override string Id => Payload.Value<string>(JwtRegisteredClaimNames.Jti) ?? String.Empty;
+        public override string Id => Payload?.Value<string>(JwtRegisteredClaimNames.Jti) ?? String.Empty;
+
+        /// <summary>
+        /// Gets the <see cref="JsonWebToken"/> associated with this instance.
+        /// </summary>
+        public JsonWebToken InnerToken { get; internal set; }
```

Comment:
It would be best if we didn't need this property.
Try and expose just Header, Payload

Reply:
Will think about this.